### ISSUE TITLE
TypeError: the JSON object must be str, not 'NoneType'

### ISSUE THREAD

**Issue by utgwkk** [tweepy/tweepy#674]
My script (using streaming API) halted with the following traceback:

```
Traceback (most recent call last):
  File "marisachan.py", line 70, in main
    stream.userstream()
  File "/home/utgw/.pyenv/versions/3.5.0/lib/python3.5/site-packages/tweepy/streaming.py", line 395, in userstream
    self._start(async)
  File "/home/utgw/.pyenv/versions/3.5.0/lib/python3.5/site-packages/tweepy/streaming.py", line 362, in _start
    self._run()
  File "/home/utgw/.pyenv/versions/3.5.0/lib/python3.5/site-packages/tweepy/streaming.py", line 295, in _run
    raise exception
  File "/home/utgw/.pyenv/versions/3.5.0/lib/python3.5/site-packages/tweepy/streaming.py", line 264, in _run
    self._read_loop(resp)
  File "/home/utgw/.pyenv/versions/3.5.0/lib/python3.5/site-packages/tweepy/streaming.py", line 325, in _read_loop
    self._data(next_status_obj)
  File "/home/utgw/.pyenv/versions/3.5.0/lib/python3.5/site-packages/tweepy/streaming.py", line 298, in _data
    if self.listener.on_data(data) is False:
  File "/home/utgw/.pyenv/versions/3.5.0/lib/python3.5/site-packages/tweepy/streaming.py", line 50, in on_data
    data = json.loads(raw_data)
  File "/home/utgw/.pyenv/versions/3.5.0/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'NoneType'
```

**Comment by svensmets** [comment]
Same error:
Traceback (most recent call last):
  File "/prog/datacollection/lib/python3.5/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/prog/datacollection/lib/python3.5/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/prog/datacollection/datacollection/twitter/tasks.py", line 125, in start_tweets_searchterms_streaming
    my_stream.filter(track=searchterms)
  File "/prog/datacollection/lib/python3.5/site-packages/tweepy/streaming.py", line 445, in filter
    self._start(async)
  File "/prog/datacollection/lib/python3.5/site-packages/tweepy/streaming.py", line 361, in _start
    self._run()
  File "/prog/datacollection/lib/python3.5/site-packages/tweepy/streaming.py", line 294, in _run
    raise exception
  File "/prog/datacollection/lib/python3.5/site-packages/tweepy/streaming.py", line 263, in _run
    self._read_loop(resp)
  File "/prog/datacollection/lib/python3.5/site-packages/tweepy/streaming.py", line 324, in _read_loop
    self._data(next_status_obj)
  File "/prog/datacollection/lib/python3.5/site-packages/tweepy/streaming.py", line 297, in _data
    if self.listener.on_data(data) is False:
  File "/prog/datacollection/lib/python3.5/site-packages/tweepy/streaming.py", line 50, in on_data
    data = json.loads(raw_data)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'NoneType'

**Comment by davidhorat** [comment]
Same error here with Tweepy 3.5.0 and Python 3.5.2:

> Traceback (most recent call last):
>   File "./twitter_daemon.py", line 223, in main
>     stream.filter(follow=user_ids)
>   File "/home/dhinvest/env/lib/python3.5/site-packages/tweepy/streaming.py", line 446, in filter
>     self._start(async)
>   File "/home/dhinvest/env/lib/python3.5/site-packages/tweepy/streaming.py", line 362, in _start
>     self._run()
>   File "/home/dhinvest/env/lib/python3.5/site-packages/tweepy/streaming.py", line 295, in _run
>     raise exception
>   File "/home/dhinvest/env/lib/python3.5/site-packages/tweepy/streaming.py", line 264, in _run
>     self._read_loop(resp)
>   File "/home/dhinvest/env/lib/python3.5/site-packages/tweepy/streaming.py", line 325, in _read_loop
>     self._data(next_status_obj)
>   File "/home/dhinvest/env/lib/python3.5/site-packages/tweepy/streaming.py", line 298, in _data
>     if self.listener.on_data(data) is False:
>   File "./twitter_daemon.py", line 122, in on_data
>     tweet = json.loads(data)
>   File "/home/dhinvest/.python3.5/lib/python3.5/json/__init__.py", line 312, in loads
>     s.__class__.__name__))
> TypeError: the JSON object must be str, not 'NoneType'
>

**Comment by fffaraz** [comment]
Same error here also with Tweepy 3.5.0 and Python 3.5.2

```
Traceback (most recent call last):
  File "/app/twitter_streaming.py", line 188, in <module>
    twitter_stream.filter(track=keywords, follow=keyauthors, stall_warnings=True)
  File "/usr/local/lib/python3.5/site-packages/tweepy/streaming.py", line 445, in filter
    self._start(async)
  File "/usr/local/lib/python3.5/site-packages/tweepy/streaming.py", line 361, in _start
    self._run()
  File "/usr/local/lib/python3.5/site-packages/tweepy/streaming.py", line 294, in _run
    raise exception
  File "/usr/local/lib/python3.5/site-packages/tweepy/streaming.py", line 263, in _run
    self._read_loop(resp)
  File "/usr/local/lib/python3.5/site-packages/tweepy/streaming.py", line 324, in _read_loop
    self._data(next_status_obj)
  File "/usr/local/lib/python3.5/site-packages/tweepy/streaming.py", line 297, in _data
    if self.listener.on_data(data) is False:
  File "/usr/local/lib/python3.5/site-packages/tweepy/streaming.py", line 50, in on_data
    data = json.loads(raw_data)
  File "/usr/local/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'NoneType'
```

**Comment by diegoje** [comment]
I'm getting the same error here using Tweepy 3.5.0 and Python 3.5.4. 

Has this error been fixed? If so, what should I do to fix it?

**Comment by Harmon758** [comment]
This should be fixed with https://github.com/tweepy/tweepy/commit/c1eddf1a5788ec4947d3a363dfe165ea599c31fc, released as part of Tweepy v3.6.0.
